db/migrate/20240820123402_copy_notifications_id_values.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class CopyNotificationsIdValues < ActiveRecord::Migration[7.0]
+  disable_ddl_transaction!
+
+  def up
+    # Short-circuit if the table has been migrated already
+    result =
+      execute(
+        "SELECT data_type FROM information_schema.columns WHERE table_name = 'notifications' AND column_name = 'id' LIMIT 1",
+      )
+    data_type = result[0]["data_type"]
+    return if data_type.downcase == "bigint"
+
+    min_id, max_id = execute("SELECT MIN(id), MAX(id) FROM notifications")[0].values
```

Comment:
```suggestion
    min_id, max_id = execute("SELECT MIN(id), MAX(id) FROM notifications WHERE new_id != id")[0].values
```

Reply:
Why is this necessary? I think it will need a table scan because `new_id` is not indexed.

Reply:
Transaction is disabled in this migration so if it fails I didn't want it to start from the beginning again. I guess I'm over worrying here since the `UPDATE` statement here checks for `new_id != id`.